addons/account/tests/test_account_move_reconcile.py

```diff
@@ -2940,18 +2977,12 @@ def test_reconcile_cash_basis_exchange_difference_transfer_account_check_entries
         ])
 
         exchange_move = receivable_line.full_reconcile_id.exchange_move_id
-        self.assertTrue(exchange_move, "There should be an exchange difference move created")
-        self.assertRecordValues(exchange_move.line_ids, [
-            {'account_id': self.cash_basis_base_account.id,                                 'debit': 0.0,   'credit': 50.0, 'amount_currency': 0.0, 'tax_ids': self.cash_basis_tax_a_third_amount.ids,  'tax_line_id': False},
-            {'account_id': self.cash_basis_base_account.id,                                 'debit': 50.0,  'credit': 0.0,  'amount_currency': 0.0, 'tax_ids': [],                                      'tax_line_id': False},
-            {'account_id': self.tax_account_1.id,                                           'debit': 0.0,   'credit': 5.0,  'amount_currency': 0.0, 'tax_ids': [],                                      'tax_line_id': self.cash_basis_tax_a_third_amount.id},
-            {'account_id': self.cash_basis_transfer_account.id,                             'debit': 5.0,   'credit': 0.0,  'amount_currency': 0.0, 'tax_ids': [],                                      'tax_line_id': False},
-        ])
+        self.assertFalse(exchange_move, "No exchange move difference should be created for the full reconcile object ,as there is no cash basis rounding.")
 
         self.assertAmountsGroupByAccount([
             # Account                               Balance     Amount Currency
             (self.cash_basis_transfer_account,      0.0,        0.0),
-            (self.tax_account_1,                    -20.0,      -10.0),
+            (self.tax_account_1,                  -15.0,      -10.0),
```

Comment:
So before creating the payment, `cash_basis_transfer_account`'s balance was `-5.0`?

I think it is nice to add a `assertAmountsGroupByAccount` between each (as well as before the first and after the last)  step of reconciliation because in the end, that is what matters

Reply:
See case 1 of the commit message. 
We had -20 because an adjustment line of -5 was created for the tax amount, in order to bring it back to the value computed from the rate used on the invoice. It's not what we want: we want to use the rate of the payment to get the domestic amount.